### ISSUE TITLE
writing guide - webview - enable svgs to embed in css

### ISSUE THREAD

**Issue by lwtchu** [openstax/cnx-recipes#5243]
was BLOCKED BY https://github.com/openstax/cnx-recipes/issues/5254 

![Screenshot 2023-09-06 at 10 09 10 AM](https://github.com/openstax/cnx-recipes/assets/11299125/95af9c1e-4674-4738-ac57-bb0abd155516)

# Brief Description of Scenario
Currently, if we bake with the web output, the SVGs are not embedded and do not show up. Please see if there's a way to embed SVGs with the delivery of the baked web generic CSS output. 

In order for baked styles for web to be adopted for Writing Guide, this must happen. This title has icons along the paragraphs that are pedagogically necessary and required.

# Acceptance Criteria
- [ ] this is just investigating options and functionality. once we figure out a way to embed SVG i can make a separate card for the work
- [ ] future proofing: this will likely apply to other SVGs such as feature icons and such if this is successful!
 

# Location(s)
- **osbooks repo:**  https://github.com/openstax/osbooks-writing-guide
- **collection slug:** writing-guide
- **ref/branch:** main
- **chapter/section:** everywhere 
- **surrounding text:**

**Comment by wilkus27** [comment]
@lwtchu I'm not sure we can say a lot about it right now. I don't know how REX and ce-styles will be connected. When we start working on that, then we can figure it out where to keep icons and how to link them between these two projects.

I added icons the same way how we do it for PDFs for now. So now they show up when we use web styles from ce-styles (by adding `&content-style=[url-to-css]` to book url). Is it ok? Should I move these changes for code review or do we want to wait with this functionality until REX and ce-styles be connected?

**Comment by lwtchu** [comment]
@wilkus27 I think we'll have to ask someone in CE 🤔 I'll bring it up during the call

**Comment by wilkus27** [comment]
### Solution from REX repo

Icons are placed in `Public` directory https://github.com/openstax/rex-web/tree/main/public/static/icons.
Then final URL is created by this function https://github.com/openstax/rex-web/blob/155a23b66df347bfd919e29a924991f56b3ad753/script/prerender/fileUtils.ts#L53

![image](https://github.com/openstax/cnx-recipes/assets/40228854/7a1fe612-0cfa-4125-9987-3e1b8e3dd2da)

Example: https://openstax.org/rex/releases/v4/a06792c/static/icons/cosmos-linguistic.svg

### Solution from PDFs

SVG is converted to data URL string https://github.com/openstax/ce-styles/blob/29195183ae6c184d9d9014f403329ffe319ed6c3/styles/build/build.dart#L19

![image](https://github.com/openstax/cnx-recipes/assets/40228854/72a56bda-6379-4896-b40b-7f718709bee4)

Usage:
![image](https://github.com/openstax/cnx-recipes/assets/40228854/67172e58-2a5d-48b7-87f1-65d0a6b2df96)

**Comment by wilkus27** [comment]
@philschatz Do you have any recommendation for this problem? Is it ok to use solution from PDFs for web styles or should we try to resolve this in a different way? Is it necessary to have URLs in a form from current solution from REX repo?

**Comment by kswanson33** [comment]
![Image](https://github.com/openstax/cnx-recipes/assets/26280712/3496b2e7-08f2-4a67-a8af-52c8d8faa9f6)

natalia's right; the way the PDFs operate, which encodes an SVG as a base64 string & uses the string as the `background` for the element in the style, should work for web. in the above screenshot see an icon from sociology embedded in the web with the following styles:

![Image](https://github.com/openstax/cnx-recipes/assets/26280712/581aa297-ca3a-4852-917b-3c1f7b14a120)

not sure where and how many icons will need to be embedded. the above styles work for inline `<p>` icons, but i imagine slightly different width/margin/etc will be needed for icons in notes (for example). 

another note is that this will add book-specific styles to `webview-generic`. this should be fine since writing guide icons seem to have unique classes, and their styles won't impact other books. just noting it since `webview-generic` has been book-agnostic in the past.

**Comment by kswanson33** [comment]
@lwtchu lmk what you think about 👆 and next steps :)

**Comment by lwtchu** [comment]
Let's try this approach where we inline embed SVG code in the CSS and just focus on the little icons in Writing Guide for now
